### PR TITLE
updating docstring for memmodel.cluster.py

### PR DESCRIPTION
**Pull request by abhsharma2**

### CODE REVIEW

File: verticapy/machine_learning/memmodel/cluster.py [hunk 1]
```diff
@@ -234,14 +237,108 @@ def transform_sql(self, X: ArrayLike) -> list[str]:
...
+
+        from verticapy.machine_learning.memmodel.cluster import KMeans
+
+    A KMeans model is defined by its cluster centroids and the p value. In this example, we will use the following:
+
+    .. ipython:: python
+            :suppress:
+
+        clusters = [[0.5, 0.6], [1, 2], [100, 200]]
+        p=2
```
Comment (oualib) [inline verticapy/machine_learning/memmodel/cluster.py:269]: follow always the same formatting logic: put a space in between the '=' sign.
p = 2

File: verticapy/machine_learning/memmodel/cluster.py [hunk 2]
```diff
@@ -234,14 +237,108 @@ def transform_sql(self, X: ArrayLike) -> list[str]:
...
+            :suppress:
+
+        model_km = KMeans(clusters, p)
+
+    Create a dataset.
+
+    .. ipython:: python
+            :suppress:
+
+        data=[[2, 3]]
```
Comment (oualib) [inline verticapy/machine_learning/memmodel/cluster.py:283]: same for this =

File: verticapy/machine_learning/memmodel/cluster.py [hunk 3]
```diff
@@ -234,14 +237,108 @@ def transform_sql(self, X: ArrayLike) -> list[str]:
...
+    **Making In-Memory Predictions**
+
+    Use :py:meth:`verticapy.machine_learning.memmodel.cluster.KMeans.predict` method to do predictions
+
+    .. ipython:: python
+            :suppress:
+
+        model_km.predict(data)[0]
+        
+    .. note:: :py:mod:`verticapy.machine_learning.memmodel.cluster.KMeans` assigns a cluster id to identify each cluster. In this example, cluster with centroid [0.5, 0.6] will have id = 0, with centroid [1,2] will have id =1 and so on.
```
Comment (oualib) [inline verticapy/machine_learning/memmodel/cluster.py:294]: same id = 1

File: verticapy/machine_learning/memmodel/cluster.py [hunk 4]
```diff
@@ -270,6 +366,97 @@ class NearestCentroid(Clustering):
...
+            :suppress:
+
+        model_nc = NearestCentroid(clusters, classes, p)
+
+    Create a dataset.
+
+    .. ipython:: python
+            :suppress:
+
+        data=[[2, 3]]
```
Comment (oualib) [inline verticapy/machine_learning/memmodel/cluster.py:404]: same

**Review by oualib — APPROVED** [review]
Looks good.

Please review the formatting of your '=' signs and it seems that your lines are too long for the black formatting.

File: verticapy/machine_learning/memmodel/cluster.py (hunk 1 above)
Comment (abhsharma2) [reply verticapy/machine_learning/memmodel/cluster.py]: done

File: verticapy/machine_learning/memmodel/cluster.py (hunk 3 above)
Comment (abhsharma2) [reply verticapy/machine_learning/memmodel/cluster.py]: done

File: verticapy/machine_learning/memmodel/cluster.py (hunk 4 above)
Comment (abhsharma2) [reply verticapy/machine_learning/memmodel/cluster.py]: done

File: verticapy/machine_learning/memmodel/cluster.py (hunk 2 above)
Comment (abhsharma2) [reply verticapy/machine_learning/memmodel/cluster.py]: done

**Comment by oualib** [comment]
@mail4umar please double check this PR and see if it is working on your side

File: verticapy/machine_learning/memmodel/cluster.py [hunk 5]
```diff
@@ -320,6 +508,120 @@ class BisectingKMeans(Clustering, Tree):
...
+
+        model_bkm.plot_tree()
+
+    .. ipython:: python
+        :suppress:
+
+        res = model_bkm.plot_tree()
+        res.render(filename='figures/machine_learning_cluster_bisecting_kmeans', format='png')
+
+    .. image:: SPHINX_DIRECTORY/figures/machine_learning_cluster_bisecting_kmeans.png
```
Comment (mail4umar) [inline verticapy/machine_learning/memmodel/cluster.py:590]: the image directive uses source as the base path. so the image path should be realtive.

.. image:: ../figures/machine_learning_cluster_bisecting_kmeans.png

**Review by mail4umar — CHANGES_REQUESTED** [review]
Just the plot_tree fix is left

**Pull request closed without merging**